### PR TITLE
LwM2M carrier library 3.3.0 modemlib test

### PR DESCRIPTION
**Pull request by nordic-hani**
Testing libcarrier 3.3.0 on top of libmodem 2.5.0

### CODE REVIEW

File: applications/serial_lte_modem/src/lwm2m_carrier/slm_at_carrier.c [hunk 1]
```diff
@@ -281,27 +288,65 @@ static int carrier_datamode_callback(uint8_t op, const uint8_t *data, int len, u
...
+		if (ret) {
+			return ret;
+		}
+
+		ret = at_params_unsigned_short_get(&slm_at_param_list, 3, &res_inst_id);
+		if (ret) {
+			return ret;
+		}
+
+		uint16_t path[4] = {19, inst_id, 0, res_inst_id};
```
Comment (stig-bjorlykke) [inline applications/serial_lte_modem/src/lwm2m_carrier/slm_at_carrier.c:335]: `LWM2M_CARRIER_OBJECT_BINARY_APP_DATA_CONTAINER`

File: applications/serial_lte_modem/src/lwm2m_carrier/slm_at_carrier.c [hunk 2]
```diff
@@ -269,7 +273,10 @@ static int carrier_datamode_callback(uint8_t op, const uint8_t *data, int len, u
 			(void)exit_datamode_handler(-EOVERFLOW);
 			return -EOVERFLOW;
 		}
-		ret = lwm2m_carrier_app_data_send(data, len);
+		uint16_t path[4] = {10250, 0, 0};
```
Comment (stig-bjorlykke) [inline applications/serial_lte_modem/src/lwm2m_carrier/slm_at_carrier.c:276]: `LWM2M_CARRIER_OBJECT_APP_DATA_CONTAINER`

File: applications/serial_lte_modem/src/lwm2m_carrier/slm_at_carrier.c [hunk 3]
```diff
@@ -281,27 +288,65 @@ static int carrier_datamode_callback(uint8_t op, const uint8_t *data, int len, u
...
 			return ret;
 		}
-	} else {
-		char data[CONFIG_SLM_CARRIER_APP_DATA_CONTAINER_BUFFER_LEN] = {0};
-		int size = CONFIG_SLM_CARRIER_APP_DATA_CONTAINER_BUFFER_LEN;
+	} else if (param_count == 3) {
+		char data[CONFIG_SLM_CARRIER_APP_DATA_BUFFER_LEN] = {0};
+		int size = CONFIG_SLM_CARRIER_APP_DATA_BUFFER_LEN;
+
+		uint16_t path[4] = {10250, 0, 0};
```
Comment (stig-bjorlykke) [inline applications/serial_lte_modem/src/lwm2m_carrier/slm_at_carrier.c:308]: `LWM2M_CARRIER_OBJECT_APP_DATA_CONTAINER`

**Comment by nordic-hani** [comment]
No longer needed, libmodem merged to main

**Pull request closed without merging**